### DISCUSSION TITLE
[Authz 插件集成 Ranger] 一些重要的xml配置文件找不到，但不影响功能

### DISCUSSION THREAD

**Discussion by fuhaiq** [apache/kyuubi#3202] [Q&A] [answered]
嗨~目前使用 Authz 插件集成 Ranger，并开启了 Kerberos，一切工作正常，存储使用 HDFS

使用 spark-sql 、spark-shell 命令，启动时会显示如下 ERROR 信息
```
root@spark-client-7d8b474c4f-jmrfv:/opt/spark/bin# ./spark-sql --deploy-mode client --conf spark.kerberos.keytab=/opt/my.keytab --conf spark.kerberos.principal=hdfs/cdh6-agent-1@TEST.COM --conf spark.kubernetes.kerberos.krb5.path=/etc/krb5.conf --conf spark.sql.extensions=org.apache.kyuubi.plugin.spark.authz.ranger.RangerSparkExtension
Setting spark.hadoop.yarn.resourcemanager.principal to hdfs
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
22/08/09 03:36:09 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
22/08/09 03:36:21 ERROR RangerConfiguration: addResourceIfReadable(ranger-spark-policymgr-ssl.xml): couldn't find resource file location
22/08/09 03:36:21 ERROR RangerConfiguration: addResourceIfReadable(ranger-spark-hive_service-audit.xml): couldn't find resource file location
22/08/09 03:36:21 ERROR RangerConfiguration: addResourceIfReadable(ranger-spark-hive_service-security.xml): couldn't find resource file location
22/08/09 03:36:21 ERROR RangerConfiguration: addResourceIfReadable(ranger-spark-hive_service-policymgr-ssl.xml): couldn't find resource file location
22/08/09 03:36:24 WARN HiveConf: HiveConf of name hive.stats.jdbc.timeout does not exist
22/08/09 03:36:24 WARN HiveConf: HiveConf of name hive.stats.retries.wait does not exist
Spark master: k8s://https://kubernetes.default:443, Application Id: spark-d0ae90d94d974ca5aff1851fb0ed556f
spark-sql> 
         > 
         > 
         > show databases;
default
tpcds_text_2
Time taken: 4.656 seconds, Fetched 2 row(s)
spark-sql>
```

提示这几个xml配置找到不到

- ranger-spark-policymgr-ssl.xml
- ranger-spark-hive_service-audit.xml
- ranger-spark-hive_service-security.xml
- ranger-spark-hive_service-policymgr-ssl.xml

其中 hive_service 是我在 Ranger Admin 中配置的服务名字

之前使用 hiveserver2 时，这几个配置文件是通过 Ranger Hive 插件里面的 `enable-hive-plugin.sh` 自动生成的。现在该如何处理呢？

**Comment by pan3793** [comment]
Please follow the doc to setup.
https://kyuubi.readthedocs.io/en/latest/security/authorization/spark/install.html#configure

**Reply by fuhaiq** [reply]
谢谢~我已经按照这个文档配置过了，其中2个xml已经在 spark/conf 目录下了
![image](https://user-images.githubusercontent.com/5200755/183562546-3926ac29-c042-40af-8f42-e544ef744f09.png)

缺失的4个xml不是这些。之前是会自动生成的

**Comment by fuhaiq** [comment, marked as answer]
搞定了~我用相同的security、audit文件代替了下。policymgr-ssl.xml 配置为空
```
<configuration>
</configuration>
```

**Comment by yaooqinn** [comment]
`spark` is the service type, `hive_service` is the service name

take  `*security.xml` for an example

the client config first loads `ranger-spark-security.xml` and then `ranger-spark-hive_service-security.xml`（only for new ranger versions, maybe 2.1 or higher).

Using one config file is enough, so the error can be dismissed

**Reply by fuhaiq** [reply]
谢谢~

**Reply by luohaifang** [reply]
ranger hive service jdbc url write “jdbc:hive2://xxx:10009/;#hive.metastore.uris=thrift://xxx:9083”  is ok? 貌似通过kyuubi不能读取到ranger policy，麻烦帮忙指点一下~

**Reply by luohaifang** [reply]
因为我的hive只有metastore
